### DISCUSSION TITLE
Parte 3: zsh

### DISCUSSION THREAD

**Discussion by gspate** [IIC2133-PUC/2024-1#145] [Tarea 2] [answered]
Hola, estoy en los tests de la parte 3 y en la parte easy y medium todo bien, pero cuando ejecuto los tests hard (1, 2 o 3) el proceso se termina porque la consola lo mata (zsh: killed). Esto será algo que ocurre en los Mac o es porque mi código está malo? (cabe decir que sí se alcanza a generar parcialmente un archivo de output, y todo lo que se alcanza a generar está igual que en el output esperado). Gracias!

**Comment by benjavicente** [comment]
Hola! 

Puede ser un problema como leak de memoria o similar, que hace que el programa sobrepase la memoria que el sistema operativo permite. 

Sube tu código para revisar cuál podría ser el problema.

**Reply by gspate** [reply]
Acabo de hacer push de mi código. Gracias!!

**Comment by benjavicente** [comment, marked as answer]
Parece ser un problema de acceso indebido de memoria al obtener los índices al hashear.

Ve si puedes resolver los errores que se ven al usar `valgrind`

```
valgrind -s ./patterns tests/patterns/easy-1/input.txt out.txt
```
